client/client.go

```diff
@@ -267,6 +269,80 @@ func vsockDial(host, port string) (net.Conn, string, error) {
 
 }
 
+func contains(elems []string, v string) bool {
+	for _, s := range elems {
+		if v == s {
+			return true
+		}
+	}
+	return false
+}
+
+// check that mdns response has all required attributes
+func mdns_required(src []string, req map[string][]string) bool {
+	txt := make(map[string]string)
+	for element := range src {
+		s := strings.SplitN(src[element], "=", 2)
+		txt[s[0]] = s[1]
+	}
+	for k, _ := range req {
+		if !contains(req[k], txt[k]) {
```

Comment:
I think you can use https://pkg.go.dev/golang.org/x/exp/slices#Contains here.  It is my understanding that this is quite possibly going to be merged into the standard library in the future.

Note that both the above package and the current implementation are O(N) which is expensive... if you use a map then it's O(1).  I don't know the size of these slices to know if that's important enough though... so I think the performance implications of the slower implementation should be ignored in favor of the simplicity of using https://pkg.go.dev/golang.org/x/exp/slices#Contains

Reply:
Thanks for the pointer.  This shouldn't be in a performance critical piece of code, so it shouldn't be a huge issue.

Reply:
So, I took your advice and everything works except for CI because we are using an older golang image.  @rminnich - can we upgrade the image we use for circleCI or should I revert back to my custom contains function?

Reply:
I need at least go 1.18 but the current image tops out at 1.17

Reply:
Had to update the underlying image for CI, but was able to follow your suggestions.